java/runtime/src/main/java/org/ray/runtime/gcs/GcsClient.java

```diff
@@ -121,17 +122,40 @@ public GcsClient(String redisAddress, String redisPassword) {
    * If the actor exists in GCS.
    */
   public boolean actorExists(ActorId actorId) {
-    byte[] key = ArrayUtils.addAll(
-        TablePrefix.ACTOR.toString().getBytes(), actorId.getBytes());
+    byte[] key =
+        ArrayUtils.addAll(TablePrefix.ACTOR.toString().getBytes(), actorId.getBytes());
+    return primary.exists(key);
+  }
+
+  public boolean wasCurrentActorReconstructed(ActorId actorId) {
+    byte[] key =
+        ArrayUtils.addAll(TablePrefix.ACTOR.toString().getBytes(), actorId.getBytes());
+    if (RAY_GCS_SERVICE_ENABLED) {
+      byte[] value = primary.get(key);
+      if (value == null) {
+        return false;
+      }
+      Gcs.ActorTableData actorTableData = null;
+      try {
+        actorTableData = Gcs.ActorTableData.parseFrom(value);
+      } catch (InvalidProtocolBufferException e) {
+        throw new RuntimeException("Received invalid protobuf data from GCS.");
+      }
+
+      long maxReconstructions = actorTableData.getMaxReconstructions();
+      long remainingReconstructions = actorTableData.getRemainingReconstructions();
+      return maxReconstructions - remainingReconstructions != 0;
+    }
```

Comment:
Ideally this can be fetched from core worker, without needing to go to gcs.

Reply:
> I agree this is not ideal. But I think it's an existing issue. We can fix this in a different PR.

Yes, I agree with u.

Reply:
@wumuzi520 @raulchen Sure. This doesn't need to be addressed in this PR. It's just something that came to my mind when looking at the code. Maybe good to add a TODO.